### PR TITLE
[12.0][ADD] new module sale delivery recreate

### PR DESCRIPTION
**Pull request by sergiocorato**

### CODE REVIEW

File: sale_delivery_recreate/readme/CONFIGURE.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Nothing todo.
```
Comment (legalsylvain) [inline sale_delivery_recreate/readme/CONFIGURE.rst]: This file is not necessary.

Comment (sergiocorato) [reply sale_delivery_recreate/readme/CONFIGURE.rst]: done, thanks!

**Comment by sergiocorato** [comment]
Runbot error should be fixed by #1278

File: sale_double_validation/models/company.py [hunk 2]
```diff
@@ -12,13 +12,13 @@ class Company(models.Model):
             ('one_step', 'Confirm sale orders in one step'),
             ('two_step', 'Get 2 levels of approvals to confirm a sale order')
         ],
-        string="Levels of Approvals",
+        string="Levels of Sale Order Approvals",
```
Comment (yostashiro) [inline sale_double_validation/models/company.py]: I guess the changes in this file are irrelevant.

Comment (sergiocorato) [reply sale_double_validation/models/company.py]: I put here the commit from #1278 to pass tests failing for these errors in main branch:
```
2020-10-05 16:21:41,823 181 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (so_double_validation, po_double_validation) of res.company() have the same label: Levels of Approvals.
2020-10-05 16:21:41,823 181 WARNING openerp_test odoo.addons.base.models.ir_model: Two fields (so_double_validation_amount, po_double_validation_amount) of res.company() have the same label: Double validation amount.
```
When merged #1278, I'll rebase.

Comment (sergiocorato) [reply sale_double_validation/models/company.py]: @yostashiro done

File: sale_delivery_recreate/__manifest__.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    "development_status": "Beta",
+    "category": "Sale",
+    "website": "https://github.com/OCA/sale-workflow",
+    "author": "Sergio Corato, "
+              "Odoo Community Association (OCA)",
+    "maintainers": ["sergiocorato"],
+    "license": "AGPL-3",
+    "installable": True,
+    "depends": [
+        "purchase_stock",
```
Comment (rousseldenis) [inline sale_delivery_recreate/__manifest__.py]: @sergiocorato Is this really needed ?

Comment (sergiocorato) [reply sale_delivery_recreate/__manifest__.py]: @rousseldenis without `purchase_stock` no buy route, so no procurement is possible and `_get_rule` gives:
`odoo.exceptions.UserError: ('No procurement rule found in location "WH/Stock" for product "[E-COM11] Cabinet with Doors".\n Check routes configuration.', '') `

Comment (rousseldenis) [reply sale_delivery_recreate/__manifest__.py]: > @rousseldenis without `purchase_stock` no buy route, so no procurement is possible and `_get_rule` gives:
> `odoo.exceptions.UserError: ('No procurement rule found in location "WH/Stock" for product "[E-COM11] Cabinet with Doors".\n Check routes configuration.', '') `

In tests or in real ?

Comment (sergiocorato) [reply sale_delivery_recreate/__manifest__.py]: @rousseldenis Ok, this is a good point. I removed the exception to include case of purchase_stock not installed, after testing with purchase_stock installed I'll launch removing it. Thanks.

Comment (sergiocorato) [reply sale_delivery_recreate/__manifest__.py]: I fixed and extended tests to cover internal pull of stock too.

Comment (sergiocorato) [reply sale_delivery_recreate/__manifest__.py]: @rousseldenis done, thanks!

**Comment by sergiocorato** [comment]
Close for inactivity

**Pull request closed without merging**